lib/Fmt_ast.ml

```diff
@@ -501,8 +501,18 @@ let rec fmt_extension c ctx key (ext, pld) =
       str s
   | ( _
     , _
-    , PStr [({pstr_desc= Pstr_value _ | Pstr_type _; _} as si)]
-    , (Pld _ | Str _ | Top) ) ->
+    , PStr
+        [ ( { pstr_desc=
+                ( Pstr_value _ | Pstr_type _ | Pstr_exception _
+                | Pstr_open {popen_override= Fresh; _}
```

Comment:
I was surprised that there was no `open!%ext M`. So I checked the parser, and actually it exists, but it needs to be spelled out `open! %ext M`. This can be fixed a separate PR if you prefer.

Reply:
No I will add it into this one

Reply:
done